### ISSUE TITLE
Include apply-changes flags options in config file

### ISSUE THREAD

**Issue by aegershman** [pivotal-cf/om#499]
see:
- https://github.com/pivotal-cf/om/tree/main/docs/apply-changes#example-yaml
- https://github.com/pivotal-cf/om/issues/494

TL;DR: would it be possible to readdress being able to set `om apply-changes`-related flags, specificially `recreate-vms:`, as a yaml field in apply-changes' `config.yml` as an option?

I rambled about an `apply-changes` options document here https://github.com/pivotal-cf/om/issues/494 but didn't really reflect on what was said in this comment https://github.com/pivotal-cf/om/issues/494#issuecomment-629475582 about the existing `apply-changes --config <config.yml>` document used for errand configuration.

## rationale 

The reason I'm bringing this up again is because recently we've had some issues that slipped by during a product upgrade that didn't involve a `recreate`; the upgrade was able to proceed just by bosh updating the package bits and cycling the associated bpm process. It turns out there were some problems that would've been noticed if we had done a full recreate of the deployments in our nonprod environments. But since we don't leave the `--recreate-vms` flag on, and we haven't yet built our own "apply-changes config file" to control whether `--recreate-vms` is passed to the apply-changes task, we ultimately didn't catch it because we didn't go through a full recreate.

If the ability to control whether `recreate-vms` was active or not via the config file, that param would snap into place with existing automation, and then it'd be just another field controlled through a git PR process like everything else.

thoughts welcome though. forgive me for resurrecting this from the other thread, but I realize what was being said with the existing `apply-changes` config file would be ... like, perfect.

thanks for your all's time 👍

**Comment by kcboyle** [comment]
@aegershman sorry we never followed up with a formal write up for this. Happy to revisit adding the other command-line args to the config file. There might be some delay (still) in us getting around to it, as we are currently focusing on different areas of Platform Automation. 
Happy to accept a PR if this is still an issue for you, though!

A couple options for this issue (open to discussion):
This is a command with an actual config file (like `configure-product`) and command line args proposed to be put into a config (like `upload-stemcell`) but it doesn't neatly fit into either category

Currently `--config` is defined as: `path to yml file containing errand configuration (see docs/apply-changes/README.md for format)`. A couple of options for what we can do here:
1. This should be changed to something like: `path to yml file for configuration (keys must match the following command line flags).`
    - we should rename the existing `config` to something like `errand-config`. This rename would be a breaking change
    - alternatively, add a new flag called `command-line-config`
      - I personally hate the way that sounds, but it would be non-breaking
    - while this can cause file fatigue, we're doing something weird here.
1. This should be changes to something like `path to yml file containing errand configuration (see docs/apply-changes/README.md for format). This may also contain command line flags`
    - this keeps a single configuration flag
    - this would have to be non-breaking
    - `apply-changes` is the only command that would act this way, and it feels weird.
      - We don't allow command line args in the config for `configure-product`, `configure-director`, etc for instance
      - `configure-authentication`, `upload-stemcell`, etc take a `config` for all command line args
    - to make this less weird, should we add `--errand-config` as a command line argument? It is a hard value to set via command line, as bash is particular about input of formatted strings, but it would make this command match existing patterns

Any other thoughts/comments?

**Comment by jaristiz** [comment]
As no comments were made after the last post, I am going to close this one and reopen if the conversation gains some traction.
